packages/unlock-js/src/KeyManager.ts

```diff
@@ -0,0 +1,167 @@
+    { name: 'lock', type: 'address' },
+    { name: 'token', type: 'uint256' },
+    { name: 'owner', type: 'address' },
+    { name: 'deadline', type: 'uint256' },
+  ],
+}
+
+export interface GetContractOptions {
+  network: number
+  signer: ethers.providers.JsonRpcSigner
+}
+export class KeyManager {
+  public networks: NetworkConfigs
+  constructor(networks?: NetworkConfigs) {
+    this.networks = networks || networkConfigs
+  }
+
+  /**
+   * This function returns the KeyManager contract for a given network.
+   */
+  getContract({ network, signer }: GetContractOptions) {
+    const networkConfig = this.networks[network]
+    const keyManagerContractAddress = networkConfig.keyManagerAddress
+    if (!keyManagerContractAddress) {
+      throw new Error('No key manager contract address found for network')
+    }
+    const KeyManagerContract = new ethers.Contract(
+      keyManagerContractAddress,
+      KeyManagerAbi,
+      signer
+    )
+    return KeyManagerContract
+  }
+
+  /**
+   * This function creates a transfer signature.
+   */
+  async createTransferSignature({
+    params,
+    signer,
+    network,
+  }: CreateTransferSignatureOptions) {
+    const networkConfig = this.networks[network]
+    const domain = {
+      name: 'KeyManager',
+      version: '1',
+      chainId: networkConfig.id,
+      verifyingContract: networkConfig.keyManagerAddress,
+    }
+    const signature = await signer._signTypedData(domain, TransferTypes, params)
+    return signature
+  }
+
+  /**
+   * This function transfers a key given a transfer signature.
+   */
+  async transfer({
+    network,
+    params: { lock, token, owner, deadline, transferSignature },
+    signer,
+  }: TransferOptions) {
+    const KeyManagerContract = this.getContract({ network, signer })
+    const tx = await KeyManagerContract.transfer(
+      lock,
+      token,
+      owner,
+      deadline,
+      transferSignature
+    )
+    return tx
+  }
+
+  /**
+   * This function sets the locksmith address.
+   */
+  async setLocksmith({
+    network,
+    params: { locksmith },
+    signer,
+  }: SetLocksmithOptions) {
+    const KeyManagerContract = this.getContract({ network, signer })
+    const tx = await KeyManagerContract.setLocksmith(locksmith)
+    return tx
+  }
+
+  /**
+   * This function checks if a transfer is possible.
+   */
+  async isTransferPossible({
+    network,
+    params: { lock, token, owner, deadline, transferSignature },
+    signer,
+  }: TransferOptions) {
+    const KeyManagerContract = this.getContract({ network, signer })
+    const tx = await KeyManagerContract.callStatic.transfer(
+      lock,
+      token,
+      owner,
+      deadline,
+      transferSignature
+    )
+    return tx
+  }
+
+  /**
+   * This function creates a wallet address from a lock address and an email address.
+   */
+  createTransferAddressWallet({
+    params: { lockAddress, email },
+  }: CreateTransferAddressKey) {
+    const privateKey = ethers.utils.keccak256(
```

Comment:
No no let's not do that because anyone could recreate the private key and impersonate the key owner.
Lets just do a hash and format as address

Reply:
ok. Created hash using id and truncated to 42 characters to work as an address.